File: 1028-interval-list-intersections/interval-list-intersections.cpp
class Solution {
public:
    vector<vector<int>> intervalIntersection(vector<vector<int>>& fl, vector<vector<int>>& sl) {
        int n = fl.size(), m = sl.size();

        int i = 0, j = 0;
        vector<vector<int>> ans;
        while(i < n && j < m){
            if(max(fl[i][0], sl[j][0]) <= min(fl[i][1], sl[j][1])){
                vector<int> cur;
                cur.push_back(max(fl[i][0], sl[j][0]));
                cur.push_back(min(fl[i][1], sl[j][1]));
                // cur.add(Math.max(fl[i][0], sl[j][0]));
                // cur.add(Math.min(fl[i][1], sl[j][1]));
                ans.push_back(cur);
            }

            if(fl[i][1] <= sl[j][1])
                i++;
            else
                j++;
        }

        return ans;
    }
};

File: 1028-interval-list-intersections/interval-list-intersections.java
class Solution {
    public int[][] intervalIntersection(int[][] fl, int[][] sl) {
        int n = fl.length, m = sl.length;

        int i = 0, j = 0;
        List<int[]> ans = new ArrayList<>();
        while(i < n && j < m){
            if(Math.max(fl[i][0], sl[j][0]) <= Math.min(fl[i][1], sl[j][1])){
                // ArrayList<Integer> cur = new ArrayList<>();
                ans.add(new int[]{Math.max(fl[i][0], sl[j][0]), Math.min(fl[i][1], sl[j][1])});
                
            }

            if(fl[i][1] <= sl[j][1])
                i++;
            else
                j++;
        }

        // return ans;
        return ans.toArray(new int[ans.size()][]);
    }
}